### PR TITLE
Use scopt to parse command line arguments.

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

**Comment by mhyee** [comment]
I was going to look at this later, and I already had a comment that you should resolve. I'll make the comments here and then open an issue or something.

**Comment by magnus-madsen** [comment]
OK; I needed the merge for something else, but just open an issue, and I will fix it.

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 1]
```diff
@@ -86,42 +89,75 @@ object Main {
...
+        text("enables the verifier.")
+
+      // Version.
+      version("version").text("prints the version number.")
+
+      // Experimental options:
+
+      // XDebug.
+      opt[Unit]("Xdebug").action((_, c) => c.copy(debug = true)).
+        text("[experimental] enables otuput of debugging information.")
```
Comment (mhyee) [inline main/src/ca/uwaterloo/flix/Main.scala]: Typo in "output"

File: main/test/ca/uwaterloo/flix/TestMain.scala [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    assert(opts.print.contains("foo"))
+  }
+
+  test("--print foo,bar") {
+    val args = Array("--print", "foo,bar", "p.flix")
+    val opts = Main.parseCmdOpts(args).get
+    assert(opts.print.contains("foo"))
+    assert(opts.print.contains("bar"))
+  }
+
```
Comment (mhyee) [inline main/test/ca/uwaterloo/flix/TestMain.scala]: Tests for -X arguments?

**Comment by mhyee** [comment]
I had some minor comments, but the main comment is to add `scopt` to `build.sbt`.

[If you go on Maven, you can easily find the library and the SBT code to add](http://search.maven.org/#artifactdetails%7Ccom.github.scopt%7Cscopt_2.11%7C3.5.0%7Cjar):

```
"com.github.scopt" % "scopt_2.11" % "3.5.0"
```

But you'll probably want to write

```
"com.github.scopt" %% "scopt" % "3.5.0"
```
which is what the other dependencies do, since `%%` means "take the Scala version specified in the SBT file"

**Comment by mhyee** [comment]
I see the fixes in master, thanks!

**Pull request merged**